File: SalesTaxes/Model/Product/Product.m
//
//  Product.m
//  SalesTaxes
//
//  Created by Fabrizio Duroni on 11/11/2016.
//  Copyright © 2016 Fabrizio Duroni. All rights reserved.
//

#import "Product.h"

@interface Product ()

/// Product name.
@property (readwrite, nonatomic, strong) NSString *productName;
/// Product price.
@property (readwrite, nonatomic, assign) float productPrice;
/// Product type.
@property (readwrite, nonatomic, assign) ProductType productType;
/// Property origin.
@property (readwrite, nonatomic, assign) ProductOrigin productOrigin;

@end

@implementation Product

- (instancetype)initWithName:(NSString *)aName andPrice:(float)aPrice andType:(ProductType)aType andOrigin:(ProductOrigin)aOrigin {
    
    self = [super init];
    
    if (self) {
        
        self.productName = aName;
        self.productPrice = aPrice;
        self.productType = aType;
        self.productOrigin = aOrigin;
    }
    
    return self;
}

@end

File: SalesTaxesSwift/Model/Product/Product.swift
//
//  Product.swift
//  SalesTaxes
//
//  Created by Fabrizio Duroni on 20/11/2016.
//  Copyright © 2016 Fabrizio Duroni. All rights reserved.
//

import Foundation

class Product {
    
    /// The name of the product.
    public private(set) var productName:String
    /// Product price.
    public private(set) var productPrice:Float
    /// Product type.
    public private(set) var productType:ProductType
    /// Property origin.
    public private(set) var productOrigin:ProductOrigin
    
    /*!
     Constructor.
     
     @param aName new product name.
     @param aPrice new product price.
     @param aType new product type.
     @param aOrigin new product origin.
     
     @returns Product instance.
     */
    init(aName:String, aPrice:Float, aType:ProductType, aOrigin:ProductOrigin) {
        
        self.productName = aName
        self.productPrice = aPrice
        self.productType = aType
        self.productOrigin = aOrigin
    }
}
